tests/job_runner_test.cc

```diff
@@ -0,0 +1,126 @@
+#include "cris/core/job_runner/job_runner.h"
+
+#include "gtest/gtest.h"
+
+#include <atomic>
+#include <chrono>
+#include <cstddef>
+#include <thread>
+
+namespace cris::core {
+
+TEST(JobRunnerTest, Basic) {
+    static constexpr std::size_t kTestBatchNum      = 3;
+    static constexpr std::size_t kThreadNum         = 4;
+    static constexpr std::size_t kJobNum            = 500;
+    static constexpr auto        kSingleJobDuration = std::chrono::milliseconds(10);
+
+    JobRunner::Config config = {
+        .thread_num_ = kThreadNum,
+    };
+    JobRunner runner(config);
+
+    auto test_batch = [&runner](const bool wait) {
+        std::atomic<std::size_t> call_count{0};
+        for (std::size_t i = 0; i < kJobNum; ++i) {
+            runner.AddJob([&call_count]() {
+                std::this_thread::sleep_for(kSingleJobDuration);
+                call_count.fetch_add(1);
+            });
+        }
+
+        if (!wait) {
+            return;
+        }
+
+        // Runner should be able to balance the loads on to the workers,
+        // so the jobs should be able to finish in about kSingleJobDuration * kJobNum / kThreadNum.
+        std::this_thread::sleep_for(kSingleJobDuration * (kJobNum * 5 / kThreadNum / 4));
```

Comment:
What if the machine is busy....

Reply:
Then we are fucked :) Do you have any better ways to check?

Reply:
What's the point of this test? Would it be covered by TLE? If so all you need is a timeout based on human tolerance, not how long the code will run.

Reply:
Then you'll need to test it in another way, maybe with some auditing worker id sequence to check the actual scheduling, and you may need to run it many times because stealing is not a requirement, it's opportunistic.

Reply:
Done